webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/web/services/EventLogDetailsGenerator.scala

```diff
@@ -786,6 +795,57 @@ class EventLogDetailsGenerator(
           }
           }
 
+        // Secret
+        case x:AddSecret =>
+          "*" #> { logDetailsService.getSecretAddDetails(x.details) match {
+            case Full(secretDiff) =>
+              <div class="evloglmargin">
+
+                { secretDetails(secretXML, secretDiff.secret)}
+                { reasonHtml }
+                { xmlParameters(event.id) }
+              </div>
+            case e:EmptyBox => logger.warn(e)
+              errorMessage(e)
+          }
+          }
+
+        case x:DeleteSecret =>
+          "*" #> { logDetailsService.getSecretDeleteDetails(x.details) match {
+            case Full(secretDiff) =>
+              <div class="evloglmargin">
+
+                { secretDetails(secretXML, secretDiff.secret)}
+                { reasonHtml }
+                { xmlParameters(event.id) }
+              </div>
+            case e:EmptyBox => logger.warn(e)
+              errorMessage(e)
+          }
+          }
+
+        case mod:ModifySecret =>
+          "*" #> { logDetailsService.getSecretModifyDetails(mod.details) match {
+            case Full(modDiff) =>
+              <div class="evloglmargin">
+
+                <h4>Secret overview:</h4>
+                <ul class="evlogviewpad">
+                  <li><b>Secret name:</b> { modDiff.name }</li>
+                </ul>
+                {(
+                "#name"  #> modDiff.name &
+                  "#value" #> mapSimpleDiff(modDiff.modValue)
```

Comment:
Should state that value has changed and (diff strucure should only have a boolean, stating it has changed or not